_components/sockets_component.h

```diff
@@ -58,11 +60,18 @@ void socket_transmitter_sta_loop(bool (*is_wifi_connected)()) {
             }
 
 #ifdef CONFIG_PACKET_RATE
-            vTaskDelay(CONFIG_PACKET_RATE != 0 ? floor(1000 / CONFIG_PACKET_RATE) : 0);
-            ets_delay_us(((1000.0 / CONFIG_PACKET_RATE) - floor(1000 / CONFIG_PACKET_RATE)) * 1000);
+            if(CONFIG_PACKET_RATE > 0) {
+                double wait_duration = (1000.0 / CONFIG_PACKET_RATE) - lag;
+                int w = floor(wait_duration);
+                vTaskDelay(w);
+                // ets_delay_us((wait_duration - w) * 1000);
+            } else
+                vTaskDelay(10);
```

Comment:
I think we can combine the two `vTaskDelay(10);` lines into one `#if` directive (See: https://stackoverflow.com/a/73538790)